src/Solvers/multigrid_solver.jl

```diff
@@ -222,13 +222,20 @@ end
 @inline create_multilevel(::RugeStubenAMG, A) = ruge_stuben(A)
 @inline create_multilevel(::SmoothedAggregationAMG, A) = smoothed_aggregation(A)
 
-function initialize_matrix(::CPU, template_field, linear_operator!, args...)
+function initialize_matrix(::CPU, template_field, linear_operator!, args...; boundary_conditions=nothing)
```

Comment:
It's a generic utility that we have so that one can create the matrix that corresponds to `linear_operator!`. The matrix can be the one used by multigrid solver or the other matrix-based solvers.

At the moment it's not used in the implicit free surface solver. The 4 nested loops were rendering this very very slow so we ended up hard-coding the matrix that corresponds to the free surface eq instead of using this utility that would work for _any_ `linear_operator!`

https://github.com/CliMA/Oceananigans.jl/blob/c21d5d6ac95e762cac85154c8f3c4de63631fca7/src/Models/HydrostaticFreeSurfaceModels/mg_implicit_free_surface_solver.jl#L87 

But I was trying to use this to get the matrix that corresponds to some other `linear_operator!` with boundary conditions other from the default for a `CenterField` and I wasn't getting what I was expecting. I realized that boundary conditions were not communicated. Not sure if a kwarg is the best solution or modifying `similar(::Field)`...

Reply:
Got it! But this PR only affects one solver?

Reply:
yes